src/Corax/IndexWriter.cs

```diff
@@ -502,31 +510,66 @@ public long Index(Slice id, Span<byte> data)
 
         private IndexedField GetDynamicIndexedField(ByteStringContext context, ref IndexEntryReader.DynamicFieldEnumerator it)
         {
-            _dynamicFieldsTerms ??= new();
+            _dynamicFieldsTerms ??= new(SliceComparer.Instance);
             using var _ = Slice.From(context, it.CurrentFieldName, out var slice);
 
             if (_dynamicFieldsTerms.TryGetValue(slice, out var indexedField))
                 return indexedField;
-            
-            
-            var clonedFieldName = slice.Clone(context);
 
-            if (_dynamicFieldsMapping != null && _dynamicFieldsMapping.TryGetByFieldName(slice, out var binding))
+            AssertIfDynamicFieldsMappingExists();
+            var persistedAnalyzer = new Lazy<ReadResult>(() =>_persistedDynamicFieldsAnalyzers.Read(slice));
```

Comment:
What guarantees the slice that in held in the `Lazy` will live long enough?

Reply:
And why lazy?

Reply:
Leftover, now it doesn't make sense :)